{
  "affected_release": null,
  "package_state": [
    {
      "product_name": "Red Hat Enterprise Linux 6",
      "fix_state": "Out of support scope",
      "package_name": "kernel",
      "cpe": "cpe:/o:redhat:enterprise_linux:6"
    },
    {
      "product_name": "Red Hat Enterprise Linux 7",
      "fix_state": "Out of support scope",
      "package_name": "kernel",
      "cpe": "cpe:/o:redhat:enterprise_linux:7"
    },
    {
      "product_name": "Red Hat Enterprise Linux 7",
      "fix_state": "Out of support scope",
      "package_name": "kernel-rt",
      "cpe": "cpe:/o:redhat:enterprise_linux:7"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "fix_state": "Out of support scope",
      "package_name": "kernel",
      "cpe": "cpe:/o:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "fix_state": "Out of support scope",
      "package_name": "kernel-rt",
      "cpe": "cpe:/o:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 9",
      "fix_state": "Fix deferred",
      "package_name": "kernel",
      "cpe": "cpe:/o:redhat:enterprise_linux:9"
    },
    {
      "product_name": "Red Hat Enterprise Linux 9",
      "fix_state": "Fix deferred",
      "package_name": "kernel-rt",
      "cpe": "cpe:/o:redhat:enterprise_linux:9"
    }
  ],
  "threat_severity": "Moderate",
  "public_date": "2025-05-01T00:00:00Z",
  "bugzilla": {
    "description": "kernel: ext4: fix warning in 'ext4_da_release_space'",
    "id": "2363510",
    "url": "https://bugzilla.redhat.com/show_bug.cgi?id=2363510"
  },
  "cvss": {
    "cvss_base_score": "",
    "cvss_scoring_vector": "",
    "status": ""
  },
  "cvss3": {
    "cvss3_base_score": "5.5",
    "cvss3_scoring_vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H",
    "status": "draft"
  },
  "iava": "",
  "cwe": "",
  "statement": "",
  "acknowledgement": "",
  "name": "CVE-2022-49880",
  "document_distribution": "",
  "details": [
    "In the Linux kernel, the following vulnerability has been resolved:\next4: fix warning in 'ext4_da_release_space'\nSyzkaller report issue as follows:\nEXT4-fs (loop0): Free/Dirty block details\nEXT4-fs (loop0): free_blocks=0\nEXT4-fs (loop0): dirty_blocks=0\nEXT4-fs (loop0): Block reservation details\nEXT4-fs (loop0): i_reserved_data_blocks=0\nEXT4-fs warning (device loop0): ext4_da_release_space:1527: ext4_da_release_space: ino 18, to_free 1 with only 0 reserved data blocks\n------------[ cut here ]------------\nWARNING: CPU: 0 PID: 92 at fs/ext4/inode.c:1528 ext4_da_release_space+0x25e/0x370 fs/ext4/inode.c:1524\nModules linked in:\nCPU: 0 PID: 92 Comm: kworker/u4:4 Not tainted 6.0.0-syzkaller-09423-g493ffd6605b2 #0\nHardware name: Google Google Compute Engine/Google Compute Engine, BIOS Google 09/22/2022\nWorkqueue: writeback wb_workfn (flush-7:0)\nRIP: 0010:ext4_da_release_space+0x25e/0x370 fs/ext4/inode.c:1528\nRSP: 0018:ffffc900015f6c90 EFLAGS: 00010296\nRAX: 42215896cd52ea00 RBX: 0000000000000000 RCX: 42215896cd52ea00\nRDX: 0000000000000000 RSI: 0000000080000001 RDI: 0000000000000000\nRBP: 1ffff1100e907d96 R08: ffffffff816aa79d R09: fffff520002bece5\nR10: fffff520002bece5 R11: 1ffff920002bece4 R12: ffff888021fd2000\nR13: ffff88807483ecb0 R14: 0000000000000001 R15: ffff88807483e740\nFS:  0000000000000000(0000) GS:ffff8880b9a00000(0000) knlGS:0000000000000000\nCS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033\nCR2: 00005555569ba628 CR3: 000000000c88e000 CR4: 00000000003506f0\nDR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000\nDR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400\nCall Trace:\n\u003cTASK\u003e\next4_es_remove_extent+0x1ab/0x260 fs/ext4/extents_status.c:1461\nmpage_release_unused_pages+0x24d/0xef0 fs/ext4/inode.c:1589\next4_writepages+0x12eb/0x3be0 fs/ext4/inode.c:2852\ndo_writepages+0x3c3/0x680 mm/page-writeback.c:2469\n__writeback_single_inode+0xd1/0x670 fs/fs-writeback.c:1587\nwriteback_sb_inodes+0xb3b/0x18f0 fs/fs-writeback.c:1870\nwb_writeback+0x41f/0x7b0 fs/fs-writeback.c:2044\nwb_do_writeback fs/fs-writeback.c:2187 [inline]\nwb_workfn+0x3cb/0xef0 fs/fs-writeback.c:2227\nprocess_one_work+0x877/0xdb0 kernel/workqueue.c:2289\nworker_thread+0xb14/0x1330 kernel/workqueue.c:2436\nkthread+0x266/0x300 kernel/kthread.c:376\nret_from_fork+0x1f/0x30 arch/x86/entry/entry_64.S:306\n\u003c/TASK\u003e\nAbove issue may happens as follows:\next4_da_write_begin\next4_create_inline_data\next4_clear_inode_flag(inode, EXT4_INODE_EXTENTS);\next4_set_inode_flag(inode, EXT4_INODE_INLINE_DATA);\n__ext4_ioctl\next4_ext_migrate -\u003e will lead to eh-\u003eeh_entries not zero, and set extent flag\next4_da_write_begin\next4_da_convert_inline_data_to_extent\next4_da_write_inline_data_begin\next4_da_map_blocks\next4_insert_delayed_block\nif (!ext4_es_scan_clu(inode, \u0026ext4_es_is_delonly, lblk))\nif (!ext4_es_scan_clu(inode, \u0026ext4_es_is_mapped, lblk))\next4_clu_mapped(inode, EXT4_B2C(sbi, lblk)); -\u003e will return 1\nallocated = true;\next4_es_insert_delayed_block(inode, lblk, allocated);\next4_writepages\nmpage_map_and_submit_extent(handle, \u0026mpd, \u0026give_up_on_write); -\u003e return -ENOSPC\nmpage_release_unused_pages(\u0026mpd, give_up_on_write); -\u003e give_up_on_write == 1\next4_es_remove_extent\next4_da_release_space(inode, reserved);\nif (unlikely(to_free \u003e ei-\u003ei_reserved_data_blocks))\n-\u003e to_free == 1  but ei-\u003ei_reserved_data_blocks == 0\n-\u003e then trigger warning as above\nTo solve above issue, forbid inode do migrate which has inline data."
  ],
  "references": [
    "https://www.cve.org/CVERecord?id=CVE-2022-49880\nhttps://nvd.nist.gov/vuln/detail/CVE-2022-49880\nhttps://lore.kernel.org/linux-cve-announce/2025050154-CVE-2022-49880-709b@gregkh/T"
  ]
}